R/epidist_utils.R

```diff
@@ -0,0 +1,795 @@
+  )
+
+  # return list of summary stats
+  list(
+    centre_spread = list(
+      mean = mean,
+      mean_ci_limits = mean_ci_limits,
+      mean_ci = mean_ci,
+      sd = sd,
+      sd_ci_limits = sd_ci_limits,
+      sd_ci = sd_ci,
+      median = median,
+      median_ci_limits = median_ci_limits,
+      median_ci = median_ci
+    ),
+    quantiles = quantiles,
+    range = list(lower_range = lower_range, upper_range = upper_range),
+    dispersion = list(
+      dispersion = dispersion,
+      dispersion_ci_limits = dispersion_ci_limits,
+      dispersion_ci = dispersion_ci
+    )
+  )
+}
+
+#' A helper function when creating an epidist object to create a summary
+#' statistics list with sensible defaults, type checking and arguments to help
+#' remember which summary statistics can be accepted in the list
+
+#' A helper function when creating an epidist object to create a citation list
+#' with sensible defaults, type checking and arguments to help remember which
+#' citation information is accepted in the list.
+#'
+#' @param author A character string of the surname of the first author. This
+#' can be underscore separated from a second author, or underscore separated
+#' from "etal" if there are more than two authors.
+#' @param year A numeric of the year of publication
+#' @param PMID A character string with the PubMed unique identifier number
+#' assigned to papers to give them a unique identifier within PubMed.
+#' @param DOI A character string of the Digital Object Identifier (DOI)
+#' assigned to papers which are unique to each paper.
+#'
+#' @return A character string of the formatted short citation
+#' @export
+#'
+#' @examples
+#' create_epidist_citation(
+#'   author = "Smith_etal",
+#'   year = 2002,
+#'   DOI = "10.19832/j.1366-9516.2012.09147.x"
+#' )
+create_epidist_citation <- function(author = NA_character_,
+                                    year = NA_integer_,
+                                    PMID = NA_character_,
+                                    DOI = NA_character_) {
+  # check input
+  checkmate::assert_character(author)
+  checkmate::assert_number(year, na.ok = TRUE)
+  checkmate::assert_number(PMID, na.ok = TRUE)
+  checkmate::assert_character(DOI)
+
+  if (is.na(author) || is.na(year) || is.na(DOI)) {
+    message(
+      "Citation cannot be created as either author, year or DOI is missing"
+    )
+    return("No citation available")
+  }
+
+  # change author formatting if multiple authors or et al
+  author <- gsub(
+    pattern = "_",
+    replacement = " ",
+    x = author,
+    fixed = TRUE
+  )
+  author <- gsub(
+    pattern = "etal",
+    replacement = "et al.",
+    x = author,
+    fixed = TRUE
+  )
+
+  # check if study has two authors and if so insert ampersand
+  num_authors <- length(unlist(strsplit(x = author, split = " ", fixed = TRUE)))
+  if (identical(num_authors, 2L)) {
+    author <- gsub(pattern = " ", replacement = " & ", x = author, fixed = TRUE)
+  }
```

Comment:
I think it may be better to enforce this formatting on the database side. I will make an issue noting this and then we can address it in it's own PR.